neutrino-custom/index.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+const { join } = require('path');
+const { ProvidePlugin } = require('webpack');
+const lint = require('./lint');
+const development = require('./development');
+const production = require('./production');
+const test = require('./test');
+const notTest = require('./not-test');
+const localWatch = require('./local-watch');
+const react = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+const entries = ['index', 'admin', 'perf', 'logviewer', 'failureviewer', 'userguide'];
+
+module.exports = neutrino => {
+    const { source } = neutrino.options;
+    const { NODE_ENV, LOCAL_WATCH } = process.env;
+
+    neutrino.config
+        // Environment-specific configuration
+        .when(NODE_ENV === 'development' && !LOCAL_WATCH, () => neutrino.use(development))
+        .when(NODE_ENV === 'production' || LOCAL_WATCH || NODE_ENV === 'test', () => neutrino.use(production))
```

Comment:
In the ui-tests Travis run chunk, we do both a `yarn test` and a `yarn run build`. From reading this it seems the test run is already very similar to the prod build. Do we still need to run both commands in the Travis run, or can we skip the `yarn run build` to save CI end to end time? Curious what is done in other projects..?

Also, it seems like we have a lot of permutations here. I have a PR locally that gets full-stack auth working with the standard development config, making local-watch.js redundant. Do you think we can get rid of test vs non-test too? It seems like there are really three configs: 1) dev, 2) test, 3) prod -- where test should either follow dev or prod rather than be a subset of both?